src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -37,7 +37,8 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
   <PropertyGroup>
-    <GenerateRuntimeConfigurationFiles Condition=" '$(GenerateRuntimeConfigurationFiles)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and ('$(HasRuntimeOutput)' == 'true' or '$(UseComHost)' == 'true') ">true</GenerateRuntimeConfigurationFiles>
+    <IsComponent Condition="'$(IsComponent)' == '' and '$(UseComHost)' == 'true'">true</IsComponent>
+    <GenerateRuntimeConfigurationFiles Condition=" '$(GenerateRuntimeConfigurationFiles)' == '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and ('$(HasRuntimeOutput)' == 'true' or '$(UseComHost)' == 'true' or '$(IsComponent)' == 'true') ">true</GenerateRuntimeConfigurationFiles>
```

Comment:
This should probably impact copy local defaults. See https://github.com/dotnet/sdk/issues/2662, which maybe this can solve. cc @peterhuene @vitek-karas

Reply:
We don't have to tie this to the initial commit, but I would love to have the end goal in mind.

Reply:
Making the project to copy nuget references by default when it opts in to being a 'component' sounds reasonable to me. Would that just be through `CopyLocalLockFileAssemblies`?